h2o-py/h2o/estimators/gbm.py

```diff
@@ -35,7 +35,7 @@ class H2OGradientBoostingEstimator(H2OEstimator):
                    "col_sample_rate_change_per_level", "col_sample_rate_per_tree", "min_split_improvement",
                    "histogram_type", "max_abs_leafnode_pred", "pred_noise_bandwidth", "categorical_encoding",
                    "calibrate_model", "calibration_frame", "custom_metric_func", "custom_distribution_func",
-                   "export_checkpoints_dir", "monotone_constraints", "check_constant_response"}
+                   "export_checkpoints_dir", "monotone_constraints", "check_constant_response", "te_model_key"}
```

Comment:
this file is auto-generated, you should not edit it manually.
For each model, the generator requests the schema for the corresponding algo and updates the `param_names` and generates the property accessors accordingly (same for the R client), so you don't need the accessors in `estimator_base.py` either.

However, for each type of model supporting TE (which means probably all of them...), you'll need to expose the property in the static `fields` property in parameters schema, see `GBMParametersV3` for example.
We should have found a way to avoid copy pasting those names, but that's the way it is today.
Although sth like this would probably work:
```
public static String[] fields = ArrayUtils.append(ModelParameters.fields, new String[] {... });
```
it would also bring some consistency to the order of those parameters (I suspect this order is then used in Flow...);

Reply:
@sebhrusen 
>this file is auto-generated, you should not edit it manually 

thanks!  missed that

>public static String[] fields = ArrayUtils.append(ModelParameters.fields, new String[] {... });

was thinking about similar thing on client side...but once it is being autogenerated it could be probably done on backend

Reply:
@sebhrusen 
> Although sth like this would probably work:
public static String[] fields = ArrayUtils.append(ModelParameters.fields, new String[] {... });

did you mean `ModelParametersSchemaV3.fields` ? Field `fields` is not static and also class `ModelParametersSchemaV3` itself is in essence abstract - so I believe we don't have `shared` part of parameters, i.e. every `xxxParametersV3` defines all the parameters itself.

Reply:
@sebhrusen noticed TargetEncoderV3 has `extractDeclaredApiParameters`, maybe I can do something like this

Reply:
Honestly, I would add it manually for now and propose this code cleanup in another ticket: whatever the approach you'll be using if you try to make this fields generation more generic, it will change the order of the fields, which is very likely to change the order of the fields in all Python generated code and maybe others too.
Without mentioning the potential impact on fields layout in Flow.

Reply:
> did you mean ModelParametersSchemaV3.fields 
yes, was pseudo-code, made the assumption that we would declare common fields in a static property there.
Relying on the introspection and `@API` annotation like it's done with `extractDeclaredApiParameters` makes more sense imo, but to ensure that the fields in Flow appear exactly in the same order, this would probably require adding an additional attribute to this annotation (sth, like `order` or `position` maybe?).

Reply:
@sebhrusen I definitely don't want to "touch" much of this in this PR. But I still think that adding fields manually on a backend will/might affect an order of fields on clients. Not sure how exactly they are being sorted, maybe just with accordance to a "natural" order in the code.

Reply:
I don't know either tbh, try one or two and see :)

Reply:
@sebhrusen btw, I've executed gen_python.py and `H2OXGBoostEstimator` got removed from some files `__init__.py` and `modelling.rst`.... it looks suspicious, am I missing something?

Reply:
@sebhrusen > btw, I've executed gen_python.py and `H2OXGBoostEstimator` got removed from some files

Probably I figured out why... it is just because XGBoost extension is not enabled in my local setup